server/oauth.js

```diff
@@ -0,0 +1,299 @@
+"use strict";
+
+var debug          = require('debug-ext')('oauth')
+  , mano           = require('mano')
+  , assign         = require('es5-ext/object/assign')
+  , userEmailMap   = require('mano/lib/server/user-email-map')
+  , login          = require('mano-auth/server/authentication').login
+  , loadToMemoryDb = require('mano/lib/server/resolve-user-access')
+  , serializeValue = require('dbjs/_setup/serialize/value')
+  , generateUnique = require('time-uuid')
+  , request        = require('request')
+  , jwtDecode      = require('jwt-decode')
+  , env            = mano.env
+  , userStorage    = mano.dbDriver.getStorage('user');
+
+var dbjsDataRecord = function (id, value) {
+	return { id: id, data: { value: serializeValue(value) } };
+};
+
+var createUser = function (data) {
+	return mano.queryMemoryDb([], 'addUser', JSON.stringify(data));
+};
+
+var updateUser = function (userId, data) {
+	var records = [];
+
+	Object.keys(data).forEach(function (key) {
+		records.push(dbjsDataRecord(userId + '/' + key, data[key]));
+	});
+
+	return userStorage.storeMany(records)(function () {
+		return loadToMemoryDb([userId]);
+	})(function () {
+		return userId;
+	});
+};
+
+var generateUrl = function (path, query) {
```

Comment:
I guess this could be general purpose util after all :)

Reply:
Propose where do you want to keep it.